inv/inventory.go

```diff
@@ -125,14 +144,26 @@ func (i *inventory) AddDevice(ctx context.Context, dev *model.Device) error {
 	if err != nil {
 		return errors.Wrap(err, "failed to add device")
 	}
+
+	i.maybeTriggerReindex(ctx, string(dev.ID))
+
 	return nil
 }
 
 func (i *inventory) DeleteDevices(
 	ctx context.Context,
 	ids []model.DeviceID,
 ) (*model.UpdateResult, error) {
-	return i.db.DeleteDevices(ctx, ids)
+	res, err := i.db.DeleteDevices(ctx, ids)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, d := range ids {
+		i.maybeTriggerReindex(ctx, string(d))
```

Comment:
We should make the workflow accept a list of device IDs to reduce the internal traffic...